kernel/src/table_changes/scan.rs

```diff
@@ -153,12 +164,34 @@ impl TableChangesScanBuilder {
             predicate: self.predicate,
             all_fields,
             have_partition_cols,
+            physical_schema: StructType::new(read_fields).into(),
         })
     }
 }
+
+impl TableChangesScan {
+    /// Returns an iterator of [`TableChangesScanData`] necessary to read CDF. Each row
+    /// represents an action in the delta log. These rows are filtered to yield only the actions
+    /// necessary to read CDF. Additionally, [`TableChangesScanData`] holds metadata on the
+    /// deletion vectors present in the commit. The engine data in each scan data is guaranteed
+    /// to belong to the same commit. Several [`TableChangesScanData`] may belong to the same commit.
+    #[allow(unused)]
+    pub(crate) fn scan_data(
```

Comment:
Isn't this the expected public entry point for CDF readers? 
Should it be public? Or do we leave it private until the feature is actually ready to use?

Reply:
Hmm I'll move it to private.